### ISSUE TITLE
Evaluating the Degree of Confidence of a Prediction.

### ISSUE THREAD

**Issue by joaopfp** [huggingface/setfit#378]
Hi, thank you for this wonderful project. When running inference I am able to see only the predicted label, it is important for me to know the degree of confidence in the prediction as well. Is there a way do to that?

**Comment by tomaarsen** [comment]
Yes. You can use `SetFitModel.predict_proba`:
https://github.com/huggingface/setfit/blob/92e966b38d295705a1cb082bf6dbec3ca5b89b76/src/setfit/modeling.py#L439

i.e.
```python
model = SetFitModel.from_pretrained(...)
results = model.predict_proba(["This movie was awful!", "I can only describe this film with one word: bliss", "That was awesome"])
probs, preds = results.max(dim=1)
```
This outputs the probabilities of each of the classes, for each of the samples. Using `max` on the `results` Tensor gives you the probabilities of the predicted class as well as the predicted classes themselves:
```
# results:
tensor([[0.9438, 0.0562],
        [0.0732, 0.9268],
        [0.0516, 0.9484]], dtype=torch.float64)
# probs:
tensor([0.9438, 0.9268, 0.9484], dtype=torch.float64)
# preds:
tensor([0, 1, 1])
```
Internally, this calls [LogisticRegression.predict_proba](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.predict_proba), which does tend to give high probabilities for the predicted class.

Hope this helps.

- Tom Aarsen
